scripts/checksum.sh

```diff
@@ -8,7 +8,12 @@ FILE=./android/app/build/outputs/apk/qa/release/app-qa-release.apk
   if test -f "$FILE"; then
     shasum -a 512 "$FILE" > ./android/app/build/outputs/apk/qa/release/sha512sums.txt
   fi
+elif [ "$MODE" == "flask" ]; then
+FILE=./android/app/build/outputs/apk/flask/release/app-flask-release.apk
 
+  if test -f "$FILE"; then
+    shasum -a 512 "$FILE" > ./android/app/build/outputs/apk/flask/release/sha512sums.txt
+  fi
 else
 FILE=./android/app/build/outputs/apk/prod/release/app-prod-release.apk
 
```

Comment:
Suggestion: parameterise the process and make the path a variable so that we can have the shasum command called only in one place with different values, using a switch case now that we have more than one non default case:
```suggestion
#!/bin/bash
MODE=$1
BASE_PATH="./android/app/build/outputs/apk"
SHA_FILE="sha512sums.txt"

case "$MODE" in
  "QA")
    DIR="qa/release"
    ;;
  "flask")
    DIR="flask/release"
    ;;
  *)
    DIR="prod/release"
    ;;
esac

FILE="${BASE_PATH}/${DIR}/app-${MODE}-release.apk"
OUTPUT_FILE="${BASE_PATH}/${DIR}/${SHA_FILE}"

if test -f "$FILE"; then
  shasum -a 512 "$FILE" > "$OUTPUT_FILE"
fi
```

Reply:
Good suggestions. I made a few modifications to your script to handle the lowercase values and set the default mode to prod. Here is my updated script.

```bash
#!/bin/bash
MODE=$1
BASE_PATH="./android/app/build/outputs/apk"
SHA_FILE="sha512sums.txt"

case "$MODE" in
  "QA")
    DIR="qa/release"
    ;;
  "flask")
    DIR="flask/release"
    ;;
  *)
    DIR="prod/release"
    MODE="prod" 
    ;;
esac

# Convert MODE to lowercase
MODE_LOWERCASE=$(echo "$MODE" | tr '[:upper:]' '[:lower:]')

FILE="${BASE_PATH}/${DIR}/app-${MODE_LOWERCASE}-release.apk"
OUTPUT_FILE="${BASE_PATH}/${DIR}/${SHA_FILE}"

if test -f "$FILE"; then
  shasum -a 512 "$FILE" > "$OUTPUT_FILE"
fi
```